File: apps/api/src/modules/auth/auth.controller.ts
/**
 * Auth Controller
 *
 * P0-1 修复：JWT 认证
 * - POST /auth/login - demo 简化登录（userId 传进来即登录，自动 upsert user）
 * - GET /auth/profile - 获取当前用户信息（需 JWT）
 *
 * R-AUTH-1 登录页面化（2026-06-28）：
 * - POST /auth/register - 注册新 ID（已存在 → 409，前端展示"该 ID 已被占用"）
 * - GET /auth/check/:userId - 检查 ID 可用性（前端实时校验）
 *   返回 { userId, available, reason? }
 *
 * ⚠️ /register 与 /login 区别：
 * - /login: 任何合规 userId 都能拿到 token（demo 临时登录 + 已存在 user 自动 upsert）
 * - /register: 严格要求 ID 不存在（创建新身份流程）
 */
import { Controller, Post, Get, Body, UseGuards, Req, Param, HttpCode, HttpStatus } from '@nestjs/common';
import { AuthService, LoginDto } from './auth.service';
import { JwtAuthGuard } from './jwt-auth.guard';

@Controller('auth')
export class AuthController {
  constructor(private auth: AuthService) {}

  /**
   * R-AUTH-1 注册新 ID
   * 已存在 → 409 Conflict
   * 格式不合法 / 保留名 → 400 Bad Request
   */
  @Post('register')
  @HttpCode(HttpStatus.OK)
  async register(@Body() dto: LoginDto) {
    return this.auth.register(dto.userId, dto.email);
  }

  /**
   * R-AUTH-1 检查 ID 可用性
   * 永远返回 200 + 结构化结果（前端实时校验用，不抛 4xx）
   */
  @Get('check/:userId')
  async check(@Param('userId') userId: string) {
    return this.auth.checkAvailability(userId);
  }

  /**
   * POST /auth/login
   * demo 简化：userId 传进来即登录（自动 upsert user）
   * 不需要密码验证
   */
  @Post('login')
  @HttpCode(HttpStatus.OK)
  async login(@Body() dto: LoginDto) {
    return this.auth.login(dto);
  }

  /**
   * GET /auth/profile
   * 获取当前登录用户信息（需要 JWT token）
   */
  @Get('profile')
  @UseGuards(JwtAuthGuard)
  async getProfile(@Req() req: any) {
    return {
      userId: req.user.userId,
      email: req.user.email,
    };
  }
}

File: apps/py/src/interview_agent/modules/auth/auth_controller.py
"""Auth Controller — 与 NestJS AuthController 像素级对齐。

路由：
- POST /auth/register — 注册新 ID（已存在 → 409）
- GET  /auth/check/:userId — 检查 ID 可用性（前端实时校验）
- POST /auth/login — demo 登录，userId 传进来即生成 token（自动 upsert user）
- GET  /auth/profile — 获取当前用户信息（需 JWT）

R-AUTH-1 登录页面化（2026-06-28）：
- /register 与 /login 区别：
  - /login: 任何合规 userId 都能拿到 token（demo 临时登录 + 已存在 user 自动 upsert）
  - /register: 严格要求 ID 不存在（创建新身份流程）

注意：
- AuthController 在 NestJS 中挂在 /auth 前缀，controller-level @Controller('auth')
- Python 端在 main.py 用 prefix='/api/auth' 挂载（与 setGlobalPrefix('api') 对齐）
"""
from fastapi import APIRouter, HTTPException, status
from pydantic import BaseModel, Field

from interview_agent.infra.db import SessionDep
from interview_agent.modules.auth import auth_service
from interview_agent.modules.auth.jwt_guard import CurrentUserDep

router = APIRouter(tags=["auth"])


class LoginRequest(BaseModel):
    userId: str = Field(..., min_length=2, max_length=50)
    email: str | None = None


@router.post("/register")
async def register(req: LoginRequest, session: SessionDep) -> dict:
    """R-AUTH-1 注册新 ID。

    返回字段与 NestJS RegisterResult 1:1：
    ```json
    { "userId", "email", "name", "created": true }
    ```
    """
    try:
        return await auth_service.register(req.userId, session, req.email)
    except auth_service.InvalidUserIdError as e:
        raise HTTPException(
            status_code=status.HTTP_400_BAD_REQUEST,
            detail=str(e),
        ) from e
    except auth_service.UserIdConflictError as e:
        raise HTTPException(
            status_code=status.HTTP_409_CONFLICT,
            detail=str(e),
        ) from e


@router.get("/check/{user_id}")
async def check_availability(user_id: str, session: SessionDep) -> dict:
    """R-AUTH-1 检查 ID 可用性（前端实时校验）。

    永远返回 200 + 结构化结果（前端用 reason 显示错误原因）：
    ```json
    { "userId": "...", "available": bool, "reason"?: "..." }
    ```
    """
    return await auth_service.check_availability(user_id, session)


@router.post("/login")
async def login(req: LoginRequest, session: SessionDep) -> dict:
    """demo 登录：userId 传进来即生成 token（自动 upsert user）。

    返回字段与 NestJS LoginResult 1:1：
    ```json
    { "accessToken", "tokenType", "expiresIn", "userId", "email", "name" }
    ```
    """
    try:
        return await auth_service.login(req.userId, session, req.email)
    except auth_service.InvalidUserIdError as e:
        raise HTTPException(
            status_code=status.HTTP_400_BAD_REQUEST,
            detail=str(e),
        ) from e


@router.get("/profile")
async def profile(current_user: CurrentUserDep) -> dict:
    """获取当前登录用户信息（需要 JWT token）。"""
    return {
        "userId": current_user["userId"],
        "email": current_user["email"],
    }
